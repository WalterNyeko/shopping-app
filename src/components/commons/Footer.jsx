import React from "react";
import { NavLink } from "react-router-dom";

const Footer = ({ departments }) => {
  return (
    <footer className="footer bg-dark sticky-bottom mb-auto py-3">
      <div className="container text-center h5 mt-4">
        {departments &&
          departments.length &&
          departments.map(({ name, department_id }) => (
            <NavLink
              to={`/department/${department_id}`}
              key={department_id}
              className="text-white mr-5"
            >
              {name}
            </NavLink>
          ))}
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-instagram fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-pinterest fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-twitter fa-3x" />
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          <i className="fab fa-facebook fa-3x" />
        </a>
      </div>

      <div className="container text-center mt-4 mb-4">
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          &copy; 2019 Shopmate Ltd
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Contact
        </a>
        <a className="text-white ml-5 mr-5 mb-4" href="#">
          Privacy policy
        </a>
      </div>
    </footer>
  );
};
export default Footer;
