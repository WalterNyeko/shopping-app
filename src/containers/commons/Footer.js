import React, { Component, Fragment } from "react";
import FooterPage from "../../components/commons/Footer";

class Footer extends Component {
  render() {
    const { departments } = this.props;
    return (
      <Fragment>
        <FooterPage departments={departments} />
      </Fragment>
    );
  }
}
export default Footer;
