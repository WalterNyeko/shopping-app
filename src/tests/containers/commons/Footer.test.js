import React from "react";
import Footer from "../../../containers/commons/Footer";
import { mount } from "enzyme";
import { Router } from "react-router";
import history from "../../../helpers/history";

describe("Footer", () => {
  const props = {
    departments: [{ department_id: 1, name: "test" }]
  };
  const wrapper = mount(
    <Router history={history}>
      <Footer {...props} />
    </Router>
  );
  it("should render footer without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
