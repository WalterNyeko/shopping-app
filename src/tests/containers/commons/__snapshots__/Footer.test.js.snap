// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render footer without crashing 1`] = `ReactWrapper {}`;
